packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -390,6 +369,318 @@ export class DataTable {
     return this.isLoading;
   }
 
+  /**
+   * Private
+   * getColumnsState function to call when updating a column state.
+   * Updating state using this function can help update the UI easily.
+   * @param state current state to modify
+   * @param stateKey key that can help identify what object in state to identify
+   * @param keyValuePairs key value pair to modify the object
+   * @returns modifies state object
+   */
+  getColumnsState(state, stateKey, keyValuePairs) {
+    let currentElementIndex;
+    const currentElement = state.filter((stateElement, elementIndex) => {
```

Comment:
can we use `findIndex` instead ? and set `currentElement = state[currentElementIndex]`

Reply:
Changed as suggested.